src/test/architecture/test_architecture.py

```diff
@@ -312,3 +312,11 @@ def test_verify_callback_uniqueness(self, mocker, caplog):
         caplog.clear()
         Architecture._verify([node_mock])
         assert len(caplog.record_tuples) == 1
+
+    def test_diff_node_names(self, mocker):
+        node_mock = mocker.Mock(spec=Architecture)
+        node_mock2 = mocker.Mock(spec=Architecture)
```

Comment:
This is not mock of NodeStruct.
Please change the naming.

ex)
- architecture_mock0